catroid/src/androidTestPhiro/java/org/catrobat/catroid/uiespresso/ui/activity/PhiroSettingsActivityTest.java

```diff
@@ -0,0 +1,172 @@
+			SETTINGS_MINDSTORMS_EV3_BRICKS_ENABLED, SETTINGS_MINDSTORMS_EV3_SHOW_SENSOR_INFO_BOX_DISABLED,
+			SETTINGS_SHOW_PARROT_AR_DRONE_BRICKS, SETTINGS_SHOW_RASPI_BRICKS, SETTINGS_CAST_GLOBALLY_ENABLED));
+
+	private Map<String, Boolean> initialSettings = new HashMap<>();
+
+	@Before
+	public void setUp() throws Exception {
+		SharedPreferences sharedPreferences = PreferenceManager
+				.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext());
+
+		for (String setting : allSettings) {
+			initialSettings.put(setting, sharedPreferences.getBoolean(setting, false));
+		}
+		initialSettings.put(SETTINGS_SHOW_PHIRO_BRICKS, sharedPreferences.getBoolean(SETTINGS_SHOW_PHIRO_BRICKS, false));
+
+		setAllSettingsTo(true);
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	private void setAllSettingsTo(boolean value) {
+		SharedPreferences.Editor sharedPreferencesEditor = PreferenceManager
+				.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext()).edit();
+
+		for (String setting : allSettings) {
+			sharedPreferencesEditor.putBoolean(setting, value);
+		}
+		sharedPreferencesEditor.apply();
+	}
+
+	@After
+	public void tearDown() {
+		SharedPreferences.Editor sharedPreferencesEditor = PreferenceManager
+				.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext()).edit();
+		for (String setting : initialSettings.keySet()) {
+			sharedPreferencesEditor.putBoolean(setting, initialSettings.get(setting));
+		}
+		sharedPreferencesEditor.apply();
+		initialSettings.clear();
+	}
+
+	@Test
+	public void basicSettingsTest() {
+		checkPreference(R.string.preference_title_enable_arduino_bricks, SETTINGS_SHOW_ARDUINO_BRICKS);
+		checkPreference(R.string.preference_title_enable_nfc_bricks, SETTINGS_SHOW_NFC_BRICKS);
+		checkPreference(R.string.preference_title_enable_hints, SETTINGS_SHOW_HINTS);
+		checkPreference(R.string.preference_title_enable_crash_reports, SETTINGS_CRASH_REPORTS);
+		checkPreference(R.string.preference_title_cast_feature_globally_enabled, SETTINGS_CAST_GLOBALLY_ENABLED);
+	}
+
+	@Test
+	public void phiroIsInitiallyActivatedTest() {
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext());
+		assertTrue("Phiro preference is not initially enabled!", sharedPreferences.getBoolean(SETTINGS_SHOW_PHIRO_BRICKS, false));
+	}
+
+	@Test
+	public void phiroIsFirstVisibleSettingTest() {
+		onData(PreferenceMatchers.withTitle(R.string.preference_title_enable_phiro_bricks))
+				.atPosition(0)
+				.check(matches(isDisplayed()));
+	}
+
+	@Test
+	public void phiroSettingsTest() {
+		onData(PreferenceMatchers.withTitle(R.string.preference_title_enable_phiro_bricks)).perform(click());
+		checkPreference(R.string.preference_title_enable_phiro_bricks, SETTINGS_SHOW_PHIRO_BRICKS);
+
+		String phiroLink = InstrumentationRegistry.getTargetContext().getString(R.string.phiro_preference_link);
```

Comment:
theres a 
public static String getResourcesString(int stringId) 

in espresso UiTestUtils that does this.